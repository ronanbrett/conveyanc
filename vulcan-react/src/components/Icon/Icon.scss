@import "src/styles/vars";

.Icon {
  border-radius: 50%;
  width: 45px;
  height: 45px;
  position: relative;

  display: flex;
  justify-content: center;
  align-items: center;

  transition: all 750ms cubic-bezier(0.075, 0.82, 0.165, 1);
}
