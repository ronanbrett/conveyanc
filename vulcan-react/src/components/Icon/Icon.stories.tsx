import React from 'react';

import Icon from './Icon';

import { action } from '@storybook/addon-actions';

export default {
  title: 'Components/Icon',
  decorators: [],
  component: Icon,
};


export const Default = (props) => (
  <Icon {...props}></Icon>
);
