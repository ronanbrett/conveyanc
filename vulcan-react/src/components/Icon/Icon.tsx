import React, { FC, forwardRef } from "react";
import classNames from "classnames";

import "./Icon.scss";

interface IconProps {
  icon: string;
  size?: "mini" | "default" | "large";
  children?: any;
}

const Icon: FC<IconProps> = forwardRef(({ icon, size, ...props }, ref) => {
  const clases = classNames({ [`Icon--${size}`]: true });
  return (
    <div ref={ref as any} className={clases}>
      <i className="material-icons">{icon}</i>
      <div className="Icon__background"></div>
    </div>
  );
});

export default Icon;
