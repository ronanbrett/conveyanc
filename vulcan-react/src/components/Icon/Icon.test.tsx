import React from 'react';
import { render } from '@testing-library/react';
import Icon from './Icon';

describe('Icon', () => {
   it('renders without error', () => {
      const { asFragment } = render(<Icon />);
      expect(asFragment()).toMatchSnapshot();
   });
});