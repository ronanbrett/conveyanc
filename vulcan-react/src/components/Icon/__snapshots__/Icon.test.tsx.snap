// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders without error 1`] = `
<DocumentFragment>
  <div
    class="Icon--undefined"
  >
    <i
      class="material-icons"
    />
    <div
      class="Icon__background"
    />
  </div>
</DocumentFragment>
`;
